torch/testing/_internal/common_utils.py

```diff
@@ -398,6 +398,18 @@ def wrapper(*args, **kwargs):
             fn(*args, **kwargs)
     return wrapper
 
+# RAII guard for setting deterministic flag and automatically resetting it
+# to its original value
+class DeterministicGuard:
+    def __init__(self, deterministic=None):
```

Comment:
I suggest removing the default value for deterministic and requiring the caller always specify whether they want deterministic on or off for readability.